### DISCUSSION TITLE
More documentation for better adoption

### DISCUSSION THREAD

**Discussion by Rodien** [oqtane/oqtane.framework#3324] [General]
Hi @sbwalker,

Thank you for all the hard work. 

What is your take on documentation? 

I believe it will significantly improve the developer experience, ultimately leading to broader adoption of the framework. However, we need experienced individuals to contribute, and others will likely follow suit.

**Comment by sbwalker** [comment]
I completely agree. A repo was set up a few years ago so that users could contribute to the documentation (https://github.com/oqtane/oqtane.docs)... however the project has not been successful in attracting technical writers. Perhaps it is because the process to create documentation is cumbersome, or perhaps it is because people are not motivated to contribute documentation to open source projects. I tend to believe the latter, as this was also a problem with the DNN project. The solution in that case was to hire a full-time content writer to create and maintain documentation as the software evolved over time ($$).

**Reply by thabaum** [reply]
I was under the impression the documentation wrote itself through documentation summaries in the code?

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/

So for example, as I create a PR for the search UI controller, I would add documentation to it, and it would get included in the Oqtane docs?!...

At least I was hoping for something that at least gave the basics, and if docs need updated we could update the source when needed.

I have contributed to Microsoft server documentation in the past.  @iJungleboy have any ideas how we can improve this area to better serve the community through contributions? 

Also I think once Oqtane version 5 comes out a lot of things that were soft or unknown about the framework should become more concrete like it was for DNN.  This would be a good topic to touch on for the upcoming developer meeting.

**Reply by Rodien** [reply]
I was actually thinking more about things like how to integrate a component library and what to look for. Step-by-step module development, like a to-do app with a database attached to it. How to build just an API, for occasions where people don't need the UI stuff.

Things like that, so developers can get up and running quickly. I mean, the more people there are who know how the platform works, really know, and have built things using the platform, the greater the chances they will have the courage to contribute to more documentation.

Or maybe, @sbwalker , you could create a playlist on YouTube on how to do the things I mentioned above (those a just some examples). I think that working on the technical aspect is great, introducing more features and so on. But educating the developers would probably increase documentation and technical pull requests as well, which in turn will relieve some burden on you.

I have not yet built anything with Oqtane, so I don't have the experience or feel that I know enough to contribute. We need to make it easier for developers to start using Oqtane.

For example, look at the documentation of ABPFramework; they have a lot of examples on how to do things.

**Reply by sbwalker** [reply]
@Rodien have you watched the module development video series: https://www.youtube.com/watch?v=JPfUZPlRRCE&list=PLYhXmd7yV0elLNLfQwZBUlM7ZSMYPTZ_f? It was created over 2 years ago and is based on the DNF.Project module which is here: https://github.com/oqtane/DNF.Projects. It stores data in a database, includes a scheduled job, integrates a third party chart component, and includes a JS Interop example.

**Reply by iJungleboy** [reply]
@sbwalker in my experience it's also the latter - people want docs, but nobody wants to write them :( it's just how this game works...
